### PR TITLE
HL-120

### PR DESCRIPTION
**Pull request by pondpiu**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-120)

- Filter Bet History Options

image
image
image
image

### CODE REVIEW

File: src/components/Bets/BetFilterGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  ["PENDING", "Pending"],
+  ["RESULTED", "Resulted"],
+  ["SETTLED", "Settled"]
+]);
+type Props = {
+  value: FilterOptions;
+  onChange: (option: FilterOptions) => void;
+  disabled: boolean;
+};
+export const BetFilterGroup = ({
```
Comment (brettvniekerk) [inline src/components/Bets/BetFilterGroup.tsx:15]: Please use `React.FC<Props>`

Comment (brettvniekerk) [reply src/components/Bets/BetFilterGroup.tsx:15]: also maybe separate the blocks of code with lines (please)

File: src/pages/Bets.tsx [hunk 2]
```diff
@@ -69,40 +32,22 @@ const Bets: React.FC = () => {
...
-              indicatorSeparator: base => ({
-                ...base,
-                display: "none"
-              })
-            }}
+        <div className="flex gap-5">
+          <BetFilterGroup
+            value={betTableFilter}
+            onChange={onFilterChange}
+            disabled={isLoading}
```
Comment (brettvniekerk) [inline src/pages/Bets.tsx:48]: why not just do `!betHistory` here

File: src/utils/bets.ts [hunk 3]
```diff
@@ -6,3 +10,37 @@ export const incrementPage = (page: number, maxPages: number) =>
...
+): BetStatus => {
+  const hasResult =
+    signedBetData.winningPropositionId || signedBetData.marketResultAdded;
+  if (!hasResult && !bet.settled) return "PENDING";
+  if (hasResult && !bet.settled) return "RESULTED";
+  if (hasResult && bet.settled) return "SETTLED";
+  throw new Error("Invalid bet status");
+};
+
+export const handleBetHistory = (
```
Comment (brettvniekerk) [inline src/utils/bets.ts:26]: this looks like formatting to me -- imo it should be in `formatting.ts` as `formatBetHistory`

File: src/utils/queries.ts [hunk 4]
```diff
@@ -1,17 +1,40 @@
...
-  limit: number,
-  skip: number,
-  address?: string
-) => `{
+const optionalFilterOptions = (filter?: FilterOptions) => {
+  switch (filter) {
+    case "ALL_BETS":
+      return "";
+    case "PENDING":
+      // TODO: filter here instead of after formatBetHistory when subgraph is updated
```
Comment (brettvniekerk) [inline src/utils/queries.ts:12]: include in comment -- "when subgraph is updated _with market oracle as data source_"

File: src/pages/Bets.tsx (hunk 2 above)
Comment (pondpiu) [reply src/pages/Bets.tsx:48]: readability

**Review by OptimusOpus — APPROVED** [review]
LGTM

**Review by brettvniekerk — APPROVED** [review]
I wish you'd use `React.FC`

**Pull request merged**